File: integration-prototypes/java/opensearch-single-rest/src/test/java/org/prototype/opensearch/single/OpenSearchSingleTests.java
package org.prototype.opensearch.single;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import org.junit.jupiter.api.Test;

public final class OpenSearchSingleTests {
    @Test
    void extractMessageFound() throws Exception {
        String payload = "{\"found\":true,\"_source\":{\"message\":\"hello opensearch single\"}}";
        assertEquals("hello opensearch single", OpenSearchSingleRepository.extractMessage(payload));
    }

    @Test
    void extractMessageNotFound() throws Exception {
        assertNull(OpenSearchSingleRepository.extractMessage("{\"found\":false}"));
    }
}

File: integration-prototypes/csharp/opensearch-single-rest/tests/OpenSearchSingleRestService.Tests/OpenSearchSingleTests.cs
using OpenSearchSingleRestService;

namespace OpenSearchSingleRestService.Tests;

public sealed class OpenSearchSingleTests
{
    [Fact]
    public void ParseMessage_ReturnsValue_WhenFound()
    {
        var value = OpenSearchSingleRepository.ParseMessage("{\"found\":true,\"_source\":{\"message\":\"hello opensearch single\"}}");
        Assert.Equal("hello opensearch single", value);
    }

    [Fact]
    public void ParseMessage_ReturnsNull_WhenNotFound()
    {
        var value = OpenSearchSingleRepository.ParseMessage("{\"found\":false}");
        Assert.Null(value);
    }
}
